subprojects/packagefiles/iir/iir/meson.build

```diff
@@ -0,0 +1,46 @@
+sources = files(
+  'Biquad.cpp',
+  'Butterworth.cpp',
+  'Cascade.cpp',
+  'ChebyshevI.cpp',
+  'ChebyshevII.cpp',
+  'Custom.cpp',
+  'PoleFilter.cpp',
+  'RBJ.cpp')
+
+headers = files(
+  'Biquad.h',
+  'Butterworth.h',
+  'Cascade.h',
+  'ChebyshevI.h',
+  'ChebyshevII.h',
+  'Common.h',
+  'Custom.h',
+  'Layout.h',
+  'MathSupplement.h',
+  'PoleFilter.h',
+  'RBJ.h',
+  'State.h',
+  'Types.h')
+
+install_headers(headers,
+                subdir: '.')
```

Comment:
Like I said in my comment above, this needs to be `subdir: 'iir'`, because Iir.h (the main header) is an "include everything" file that then does:

```c
#include "iir/Common.h"
#include "iir/Biquad.h"
#include "iir/Cascade.h"
#include "iir/PoleFilter.h"
#include "iir/State.h"
#include "iir/Butterworth.h"
#include "iir/ChebyshevI.h"
#include "iir/ChebyshevII.h"
#include "iir/Custom.h"
#include "iir/RBJ.h"
```

So the component files need to be installed to a subdir next to Iir.h (mirroring the layout in the source tree, actually).

Reply:
That's confusing. 

If non-install calls operate on files relative to the meson.build file, why would install calls not do the same?

It appears install actions drop this tree relationship awareness.

I'll add this subdirectory, none the less.

Thanks for explaining this.

Reply:
For the same reason any other install calls, simply install directly to whichever install_dir they have.

executable installs to `get_option('bindir')`, even if it is in src/meson.build
library installs to `get_option('libdir')`, even if it is in lib/meson.build

So too, install_headers() installs to `get_option('includedir')`, even if it is in iir/meson.build

But install_headers() has a common requirement to install to "includedir, but a subdirectory of it", which is why it has a `subdir:` option in addition to `install_dir:`

We *cannot* automatically install to iir/, because often people have all their includes in include/meson.build, and it would be wrong to install them to `get_option('includedir') / 'include'`.

:)

Reply:
Also, can you please remove `subdir: '.'` from the other one? :) It does nothing.

Reply:
Done.